optaplanner-core/src/main/java/org/optaplanner/core/config/util/ConfigUtils.java

```diff
@@ -401,8 +402,9 @@ public static int resolvePoolSize(String propertyName, String value, String... m
                     + PlanningId.class.getSimpleName() + " annotation.");
         }
         Member member = memberList.get(0);
-        MemberAccessor memberAccessor = MemberAccessorFactory.buildMemberAccessor(member, FIELD_OR_READ_METHOD,
-                PlanningId.class);
+        MemberAccessor memberAccessor =
+                MemberAccessorFactory.buildMemberAccessor(domainAccessType, member, FIELD_OR_READ_METHOD,
```

Comment:
Method signature: domainAccessType is less important to understand this code than member and FIELD_OR_READ_METHOD and even the PlanningId.class. Let's make that the last parameter.